### DISCUSSION TITLE
mpd Will Not Start

### DISCUSSION THREAD

**Discussion by teslasparks301** [rern/rAudio#1139] [General]
I have been using rAudio1 for a few years now and have been very pleased with it. However, the last update, 20221122 (maybe 20221123), introduced a problem where mpd will not start using systemd. Trying to work through problem I found that if I used my old mpd.conf versus the new format it would work. The net differences are what I would consider to be minor. The main difference is that the new format uses 'include_optional ' statements while my old conf file explicitly defines parameters in the file itself. I have been working on this for several days and am at the point of losing my sanity. rAudio shows this when mpd fails to start:

`mpd 0.23.11-1
# /srv/http/data/mpdconf/mpd.conf
audio_buffer_size       "49152"
auto_update             "yes"
bind_to_address         "0.0.0.0"
db_file                 "/srv/http/data/mpd/mpd.db"
log_file                "/var/log/mpd.log"
log_level               "verbose"
max_output_buffer_size  "49152"
music_directory         "/mnt/MPD"
playlist_directory      "/srv/http/data/playlists"
state_file              "/var/lib/mpd/mpdstate"
user                    "mpd"
volume_normalization    "yes"
input {
	plugin  "curl"
}
audio_output {
	name        "Snapcast"
	type        "fifo"
	path        "/tmp/snapfifo"
	format      "48000:16:2"
	mixer_type  "software"
}
resampler {
	plugin   "soxr"
	quality  "very high"
	thread   "1"
}
audio_output {
	name           "HiFiBerry DAC+"
	device         "hw:0,0"
	type           "alsa"
	auto_resample  "no"
	auto_format    "no"
	mixer_type     "hardware"
	mixer_control  "Digital"
	mixer_device   "hw:0"
}

mpd.service- Music Player Daemon
     Loaded: loaded (/usr/lib/systemd/system/mpd.service; disabled; preset: disabled)
    Drop-In: /usr/lib/systemd/system/mpd.service.d
             └─00-arch.conf
             /etc/systemd/system/mpd.service.d
             └─override.conf
     Active: failed (Result: timeout) since Mon 2022-12-12 12:36:50 EST; 16min ago
       Docs: man:mpd(1)
             man:mpd.conf(5)
    Process: 719 ExecStart=/usr/bin/mpd --systemd /srv/http/data/mpdconf/mpd.conf (code=killed, signal=KILL)
   Main PID: 719 (code=killed, signal=KILL)
        CPU: 23.277s

Dec 12 12:33:55 rAudio mpd[719]: Dec 12 12:33 : curl: version 7.86.0
Dec 12 12:33:55 rAudio mpd[719]: Dec 12 12:33 : curl: with OpenSSL/3.0.7
Dec 12 12:35:19 rAudio systemd[1]: mpd.service: start operation timed out. Terminating.
Dec 12 12:36:50 rAudio systemd[1]: mpd.service: State 'stop-sigterm' timed out. Killing.
Dec 12 12:36:50 rAudio systemd[1]: mpd.service: Killing process 719 (mpd) with signal SIGKILL.
Dec 12 12:36:50 rAudio systemd[1]: mpd.service: Killing process 721 (n/a) with signal SIGKILL.
Dec 12 12:36:50 rAudio systemd[1]: mpd.service: Main process exited, code=killed, status=9/KILL
Dec 12 12:36:50 rAudio systemd[1]: mpd.service: Failed with result 'timeout'.
Dec 12 12:36:50 rAudio systemd[1]: Failed to start Music Player Daemon.`

My old format mpd.conf is:

`auto_update            "no"
max_output_buffer_size "49152"
audio_buffer_size      "49152"
bind_to_address        "0.0.0.0"
db_file                "/srv/http/data/mpd/mpd.db"
music_directory        "/mnt/MPD"
playlist_directory     "/var/lib/mpd/playlists"
replaygain             "off"
state_file             "/var/lib/mpd/mpdstate"
user                   "mpd"
volume_normalization   "yes"
log_file               "/var/log/mpd.log"
#log_level              "verbose"

input {
	plugin         "curl"
}

resampler {
	plugin         "soxr"
	quality        "very high"
}

audio_output {
	name           "HiFiBerry DAC+"
	device         "hw:0,0"
	type           "alsa"
	auto_resample  "no"
	auto_format    "no"
	mixer_type     "software"
}

audio_output {
	name           "Snapcast"
	type           "fifo"
	path           "/tmp/snapfifo"
	format         "48000:16:2"
	mixer_type     "software"
}`

The log, /var/log/mpd.log, shows nothing.

`Dec 12 10:54 : state_file: Loading state file /var/lib/mpd/mpdstate
Dec 12 10:54 : inotify: initializing inotify
Dec 12 11:00 : state_file: Loading state file /var/lib/mpd/mpdstate
Dec 12 11:00 : inotify: initializing inotify
Dec 12 11:19 : state_file: Loading state file /var/lib/mpd/mpdstate
Dec 12 11:19 : inotify: initializing inotify
Dec 12 12:07 : state_file: Loading state file /var/lib/mpd/mpdstate
Dec 12 12:07 : inotify: initializing inotify
Dec 12 12:33 : state_file: Loading state file /var/lib/mpd/mpdstate
Dec 12 12:33 : inotify: initializing inotify`

I did a fresh install using the release of i20221117. Still the same. Guidance is sorely needed. The work-around is to just use my old mpd.conf but many feature changes done via the web interface are done using include statements. That means I would not see the changes.

**Comment by rern** [comment]
Start with default Player settings then set these custom values:
- `audio_buffer_size` - set with `Buffer - Audio`
- `max_output_buffer_size` set with `Buffer - output`
- `log_file` - set with `User's Configurations`
    - `mpd` cannot write to directory `/var/log/` which will result in failed start.

**Comment by teslasparks301** [comment]
I did what you said and it started working. Then I started adding other options and there was still much joy. Then I set my last option, auto update, and it failed. I always wanted to change that to a yes in my hard-coded conf file but kept forgetting. Had I done it, I wouldn't have had to bother you. It was one of the differences I thought was trivial. Oh, the ego! Live and learn.

So I guess because my library is so large, inotify is not getting back to systemd in the allotted time to let it know that mpd started. I realize that this is a mpd problem and not a rAudio problem. rAudio is still the best rpi player out there.
